### PR TITLE
[WWST-2298] Add Sengled Door Sensor device to existing DTH

### PR DESCRIPTION
**Pull request by MMateusiakS**
//01 0104 0402 00 07 0000 0001 0003 0020 0500 0B05 FC02 00
Because of device capabilities it suits well in Nyce DTH. 

Waiting for your feedback @MAblewiczS

### CODE REVIEW

File: devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy [hunk 1]
```diff
@@ -353,9 +356,8 @@ Integer convertHexToInt(hex) {
 }
 
 def refresh() {
-	log.debug "Refreshing Battery"
 	def refreshCmds = [
-			"st rattr 0x${device.deviceNetworkId} ${endpointId} 1 0x20", "delay 200"
+		"st rattr 0x${device.deviceNetworkId} ${endpointId} 1 0x20", "delay 200"
```
Comment (tpmanley) [inline devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: Could you also update the DTH to use the `zigbee` library wherever possible, for example replace this with `zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020)`

Comment (MMateusiakS) [reply devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: Sure,  Tom. I will work on it.

Comment (MMateusiakS) [reply devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: I need so more work not to let device to be disconnected.

**Comment by MMateusiakS** [comment]
@tpmanley can you review commits? I am bit afraid of futher refactoring of NYCE DTH. Also I took one Nyce open/close sensor and checked if my changes work on it. Changes work on sengled and on nyce device. But I cannot receive "pings". Tom, what can be wrong if I do not get any "ping" in IDE?

**Comment by tpmanley** [comment]
> @tpmanley can you review commits? I am bit afraid of futher refactoring of NYCE DTH. Also I took one Nyce open/close sensor and checked if my changes work on it. Changes work on sengled and on nyce device. But I cannot receive "pings". Tom, what can be wrong if I do not get any "ping" in IDE?

pings from the nyce or the sengled? The nyce sensors are known to be very sleepy so it's unlikely they will respond to pings.

File: devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy [hunk 2]
```diff
@@ -299,51 +259,18 @@ def configure() {
...
-	]
+	if(device.getDataValue("manufacturer") == "sengled") {
+		return zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) + zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) + zigbee.batteryConfig(30, 300) + zigbee.enrollResponse()
+	} else {
+		// battery minReportTime 30 seconds, maxReportTime 5 min. Reporting interval if no activity
+		return zigbee.iasZoneConfig() + zigbee.batteryConfig(30, 300) + refresh() // send refresh cmds as part of config
+	}
 }
 
 private hex(value) {
```
Comment (tpmanley) [inline devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: I think this is unused now and can be removed

File: devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy [hunk 3]
```diff
@@ -299,51 +259,18 @@ def configure() {
...
-		array[j] = array[i];
-		array[i] = tmp;
-		j--;
-		i++;
-	}
-
-	return array
-}
-
 private getEndpointId() {
```
Comment (tpmanley) [inline devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: This also appears unused

**Comment by tpmanley** [comment]
@greens can take a look too, but I think the changes made so far look good

File: devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy (hunk 2 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: sure, removed

File: devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy (hunk 3 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/nyce-open-closed-sensor.src/nyce-open-closed-sensor.groovy]: sure, deleted

**Review by tpmanley — APPROVED** [review]

**Comment by MMateusiakS** [comment]
> > @tpmanley can you review commits? I am bit afraid of futher refactoring of NYCE DTH. Also I took one Nyce open/close sensor and checked if my changes work on it. Changes work on sengled and on nyce device. But I cannot receive "pings". Tom, what can be wrong if I do not get any "ping" in IDE?
> 
> pings from the nyce or the sengled? The nyce sensors are known to be very sleepy so it's unlikely they will respond to pings.

I have this issue with sengled. I removed battery and if I don't get any pings, device is not going to be "disconnected".

**Comment by greens** [comment]
That might be more of an issue with the Device Health service than it is with this device. I'd just verify that you're seeing the pings be sent in the events list. These changes lgtm.

**Comment by MMateusiakS** [comment]
Today I see pings in IDE :) So, no DTH issue.

**Review by greens — APPROVED** [review]

**Pull request merged**